connectors/sources/sharepoint_server.py

```diff
@@ -483,6 +478,7 @@ def get_default_configuration(cls):
                 "label": "Comma-separated list of SharePoint site collections to index",
                 "order": 4,
                 "type": "list",
+                "required": False
```

Comment:
Why make it false? Empty value will be treated like "index only root site"?

Reply:
This is my mistake, I am planning to add a configuration checkbox that allows the connector to try to get the list of sites from SharePoint Search, making this field optional.

Reply:
I see - it's possible, but I'd advice to make it in a separate PR so that we could review it better. Large PRs with multiple features tend to take a lot of time to review, fix and merge